### DISCUSSION TITLE
Server Side Validation

### DISCUSSION THREAD

**Discussion by rboklage** [quasarframework/quasar#11005] [General - Components / Directives / etc]
I'm new to VueJs and Quasar and would like to use these in an existing web app. I have a form that is validated server side and if there is an error a message is returned based on the user's language. In previous apps where I used jQuery I would loop through the errors and add the error message to the DOM.

I'm guessing I can do the same with q-input, however I can't figure out how to update the error-message and trigger :error to true. 

<q-input square clearable v-model.lazy="form.fname" type="text" label="First Name" ref="fname" error-message=""></q-input>
this.$refs.fname.$q.error = true;

**Comment by pikil** [comment]
You can bind the error as a variable to the element in the template, like below (assuming, you are using Quasar 1):

```
<template>
  <q-input v-model="fName" :error="errors.fName" error-message="First name is incorrect..." />
</template>
```

```
<script>
export default {
  data () {
    return {
      fName: '', // JS variable for first name
      errors: {
        fName: false
      }
    }
  }
}

</script>
```

* It is better if you stick to Quasar 2, though :)

**Reply by rboklage** [reply]
I was planning on using Quasar 2. So how would I change the error-message that is returned from an ajax call?

**Reply by pikil** [reply]
```
<template>
  <q-input v-model="fName" :error="errors.fName.status" :error-message="errors.fName.msg" />
</template>
```

```
<script>
import { defineComponent, ref } from 'vue'

export default defineComponent({
  setup () {
    const errors = ref({
      fName: {
        status: false,
        msg: ''
      }
    })

    return {
      errors
    }
  }
})
</script>
```

Now you can change the text by doing something like this in JS: ```errors.value.fName.msg = 'Incorrect first name...'```

**Reply by rboklage** [reply]
Thanks for your help, Still not sure if I'm going to use this. I get what you say and that does work, however if I set the status to true and there is no value in the field the error does not display. If the field is required there should be a message. Would I also need to use q-form to trigger validation?

**Reply by pikil** [reply]
Could you please create codepen or jsfiddle example so I or others can have a look?

**Reply by rboklage** [reply]
I have set up a code snippet of my registration form. Note that is going to be used in an existing app so I'm using CDN for both VueJs and Quasar. After clearing each message there would be an Ajax call and if there where errors I would loop through the fields and set the messages

https://jsfiddle.net/1qw36n0h/

**Reply by pikil** [reply]
That's just because you've mixed options API and composition API. If you move `errors` inside `data()`, that will be working you expect ;)

```
      data() {
            return {
              form: {
                fname: '',
                lname: '',
              },
              errors: {
                fname: {
                  err: false,
                  msg: ''
                },
                lname: {
                  err: true,
                  msg: 'Incorrect last name'
                }
              }
            }
          }
```

* However, Options API is related to Vue2, whereas Composition API is related to Vue3.

**Reply by rboklage** [reply]
Thanks for your help. I have updated the code to make use of Composition API. Seems to work now. I have implemented this in my web app
